challenges/Solution 2/TestProject/src/restApi/RestApiCall.java

```diff
@@ -0,0 +1,94 @@
+package restApi;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.sun.jersey.multipart.FormDataParam;
+
+@Path("/rest")
+public class RestApiCall {
+
+	private static List<Integer> numberListInt;
+	private static final int expectedListSize = 500;
+
+	@POST
+	@Consumes(MediaType.MULTIPART_FORM_DATA)
+	@Path("/data")
+	public String validateNumbersPost(@FormDataParam("file") InputStream fileInputStream) {

```

Comment:
I know you mentioned you are new to rest APIs. As a suggestion in the future, consider using the Spring Rest Controller inside Java. It manages much of the API and data handling for you. Then you could just have an input of List<String> numberList and Spring would give automatically translate the JSON data into the Java object for you or return an error to client if it doesn't match.